gc/base/GCExtensionsBase.hpp

```diff
@@ -906,6 +906,10 @@ class MM_GCExtensionsBase : public MM_BaseVirtual {
 	MMINLINE OMR_VM* getOmrVM() { return _omrVM; }
 	MMINLINE void setOmrVM(OMR_VM* omrVM) { _omrVM = omrVM; }
 
+#if defined(OMR_GC_MODRON_SCAVENGER)
+	virtual void registerScavenger(MM_Scavenger *scavenger) { this->scavenger = scavenger; }
+	virtual void unregisterScavenger() { scavenger = NULL; }
+#endif /* defined(OMR_GC_MODRON_SCAVENGER) */
```

Comment:
sorry for the late observation, but I'm thinking to pull the #ifdefs inside the methods.
It's a mixed bag how we do it accross GC code, but for Scavenger related methods and some other helpers inside this class, we tend to keep them inside.

An example is:

```
	MMINLINE bool
	isScavengerEnabled()
	{
#if defined(OMR_GC_MODRON_SCAVENGER)
		return scavengerEnabled;
#else /* defined(OMR_GC_MODRON_SCAVENGER) */
		return false;
#endif /* defined(OMR_GC_MODRON_SCAVENGER) */
	}
````

Reply:
and move them closer to isScavengerEnabled()